PSPDFKit/react-native
# Add callback for when an annotation is added, changed or deleted

iOS implementation for #90 

---

## How to test

* Create a new React Native sample app: https://github.com/PSPDFKit/react-native#ios
* For step 4, use this branch: `yarn add github:PSPDFKit/react-native#rad/add-onAnnotationTapped-ios`
* For step 15, use the following code in your `App.js`

```javascript
import React, { Component } from 'react';
import { NativeModules } from 'react-native';
import PSPDFKitView from 'react-native-pspdfkit'

var PSPDFKit = NativeModules.PSPDFKit;
PSPDFKit.setLicenseKey('YOUR_LICENSE_KEY_GOES_HERE');

export default class App extends Component<{}> {
  render() {
    return (
      <PSPDFKitView
        document={'document.pdf'}
        configuration={{
          pageTransition: 'scrollContinuous',
          scrollDirection: 'vertical',
          documentLabelEnabled: true,
        }}
        style={{ flex: 1, color: '#267AD4' }}
        onAnnotationsChanged={this.onAnnotationsChanged}      
      />
    )
  }
  onAnnotationsChanged(event) {
      if (event['error']) {
          alert(event['error']);
      } else {
          alert('Annotations ' +event['change'] + ': ' + JSON.stringify(event['annotations']));
      }
  }
}
``` 

* Launch the app
* Add a new annotation and notice the alert.
* Exit the annotation mode by closing the annotation toolbar
* Move the recently added annotation and observe the alert
* Delete the annotation and observe the alert

![recording](https://user-images.githubusercontent.com/7443038/42952720-1761b506-8b47-11e8-937a-0654926b94fe.gif)

## What to test:

- [ ] Make sure that there are now issues on Android and Windows
